pkg/tls/tlsmanager.go

```diff
@@ -113,12 +124,16 @@ func (m *Manager) Get(storeName string, configName string) (*tls.Config, error)
 			}
 
 			if cert != nil {
+				m.updateMetrics(cert)
 				return cert, nil
 			}
 		}
 
 		bestCertificate := store.GetBestCertificate(clientHello)
+
 		if bestCertificate != nil {
+			m.updateMetrics(bestCertificate)
+
```

Comment:
Hello @sylr,

Here, i think the updateMetrics calls must not be done on the critical path more appropriately each time the certificate_store stores a certificate (L84), since the metric should change only when the certs are updated.
It also has to be call when configuring the default certificate if it is not generated.

WDYT ?

Reply:
Hi @rtribotte, sorry for the late reply byt I'm currently on vacation.

I'll review your proposition when I get back.

Thanks.